### PR TITLE
issue-120: Adding STO compat checks

### PR DESCRIPTION
**Pull request by dartov**

### CODE REVIEW

**Comment by dartov** [comment]
Tested locally, works

File: model_definitions/dfd4052e-f91b-4aa5-9c79-f26d649dd931/model_modules/training.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def train(data_conf, model_conf, **kwargs):
                    password=os.environ["AOA_CONN_PASSWORD"],
                    database=data_conf["schema"] if "schema" in data_conf and data_conf["schema"] != "" else None)
 
+    check_sto_version()
     cleanup_cli(model_version)
```
Comment (willfleury) [inline model_definitions/dfd4052e-f91b-4aa5-9c79-f26d649dd931/model_modules/training.py:29]: maybe add the table name argument so its clear. In fact we can probably have the table name defined here at the top as we use it later in the code when we save the models also?

File: model_definitions/dfd4052e-f91b-4aa5-9c79-f26d649dd931/model_modules/training.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def train(data_conf, model_conf, **kwargs):
                    password=os.environ["AOA_CONN_PASSWORD"],
                    database=data_conf["schema"] if "schema" in data_conf and data_conf["schema"] != "" else None)
 
+    check_sto_version()
```
[the remark this answers is not in the document]
Comment (dartov) [reply model_definitions/dfd4052e-f91b-4aa5-9c79-f26d649dd931/model_modules/training.py:28]: Good catch, thanks!

File: model_definitions/dfd4052e-f91b-4aa5-9c79-f26d649dd931/model_modules/training.py (hunk 1 above)
Comment (dartov) [reply model_definitions/dfd4052e-f91b-4aa5-9c79-f26d649dd931/model_modules/training.py:29]: Sure

**Comment by dartov** [comment]
@willfleury what if we put STO models table name in model_conf for STO models? It would allow users to configure it for each model version (and have that default exposed), and once model version is trained - it won't change for evaluation and scoring, so we can just read it there

**Comment by willfleury** [comment]
> @willfleury what if we put STO models table name in model_conf for STO models? It would allow users to configure it for each model version (and have that default exposed), and once model version is trained - it won't change for evaluation and scoring, so we can just read it there

I think the default for this needs to be a project level override but with a global default. The same goes for the ivsm table etc and it also needs to support specifying the database. Users should also be able to override for eg testing but I think that should be the exception.

**Pull request merged**